src/NuGet.Clients/NuGet.SolutionRestoreManager/NuGetSolutionService.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ComponentModelHost;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.SolutionRestoreManager
+{
+    internal class NuGetSolutionService : INuGetSolutionService
+    {
+        [Import]
+        internal Lazy<ISolutionRestoreWorker> SolutionRestoreWorker { get; set; }
+
+        private bool _initialized = false;
+
+        public async Task RestoreSolutionAsync(CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            await EnsureInitializedAsync();
+            await SolutionRestoreWorker.Value.ScheduleRestoreAsync(SolutionRestoreRequest.ByMenu(), cancellationToken);
+        }
+
+        private async Task EnsureInitializedAsync()
```

Comment:
Can we [AsyncLazyInitializer.InitializeAsync](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.threading.asynclazyinitializer.initializeasync?view=visualstudiosdk-2019#Microsoft_VisualStudio_Threading_AsyncLazyInitializer_InitializeAsync_System_Threading_CancellationToken_) method?

Reply:
sure, we can do that. 
We haven't used it before but a good time to start.